monkey-patch.js

```diff
@@ -88,8 +88,16 @@ function patchAudioContext(NativeAudioContext) {
 function patchOfflineAudioContext(NativeOfflineAudioContext) {
   class OfflineAudioContext extends NativeOfflineAudioContext {
     constructor(...args) {
-      super(...args);
-      console.log(args);
+      // handle initialisation with either an options object or a sequence of parameters
+      // https://developer.mozilla.org/en-US/docs/Web/API/OfflineAudioContext/OfflineAudioContext#parameters
+      if( typeof args[0] === 'object'
+          && 'numberOfChannels' in args[0] && 'length' in args[0] && 'sampleRate' in args[0]
```

Comment:
Would be nice to check each of the parameter and throw a NotSupportedError if something is not right
> A [NotSupportedError](https://webidl.spec.whatwg.org/#notsupportederror) exception MUST be thrown if any of the arguments is negative, zero, or outside its nominal range.

Error handling is still in the todo list on the rust side, but we will have in any case something crashing with `NumberExpected` which is not very informative, if we do something like:
```
const offline = new OfflineAudioContext({
  sampleRate: 0,
});
```

Reply:
Now throwing that error if the constructor values are not positive integers (JS has no concept of long) or positive floats.  Are there any nominal range checks to perform here?

Reply:
> Are there any nominal range checks to perform here?

Yep but if we make sure we go into the rust constructor properly, I think this can be delegated to rust so we don't have to duplicate the values and make the checks twice.

My concern was more about that writing:
```js
const offline = new OfflineAudioContext({ sampleRate: 0, });
```
can't go into the else branch, which is still the case right now (but my comment was quite misleading sorry)

Maybe doing something like:
```js
if (typeof args[0] === 'obj') {
   const options = args[0];
   if ('numberOfChannels' in options && 'length' in options && 'sampleRate' in options) {
      // here we can delegate the error handling to rust (which is still in the todo list...)
      super(numberOfChannel, length, sampleRate);
   } else {
       throw Error('NotSupportedError - Invalid options, must contain blah, blah...');
   }
} else {
   super(...args);
}
```
would work?

Reply:
> JS has no concept of long

Just to note, JS Number are 64 bits, what we actually don't have are floats :)

Reply:
Actually forget my comment, I'm wrong here... `undefined` is not a `positiveInt`